### PR TITLE
RELATED: BB-1690 Support measure value filters in UI SDK

### PR DESCRIPTION
**Pull request by qduval**

### CODE REVIEW

**Comment by yenkins** [comment]
Currently destined for [release R191](https://confluence.intgdc.com/display/Dev/R191).
This release is **zero downtime** release.
Code drop: 2019-09-20 00:00.

**Comment by yenkins** [comment]
Story [BB-1690](http://jira.intgdc.com/browse/BB-1690) Support measure value filters in UI SDK
	Status: Hacking
	Priority: Major
	Assignee: Quentin Duval

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/41/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/115/ : SUCCESS in 1m 29s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-check/42/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/116/ : SUCCESS in 1m 00s

File: src/AFM.ts [hunk 1]
```diff
@@ -150,6 +151,45 @@ export namespace AFM {
...
+    export interface ILocalIdentifierQualifier {
+        localIdentifier: string;
+    }
+
+    export type Qualifier = ObjQualifier | ILocalIdentifierQualifier;
+
+    export interface IMeasureValueFilter {
+        measureValueFilter: {
+            measure: Qualifier;
+            condition?: MeasureValueFilterCondition;
```
Comment (strouhaljan) [inline src/AFM.ts]: How do we work with the measure value filter when it has no condition?

File: src/VisualizationObject.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ export namespace VisualizationObject {
 
     export type BucketItem = IMeasure | IVisualizationAttribute;
 
+    export type VisualizationObjectExtendedFilter = VisualizationObjectFilter | IMeasureValueFilter;
```
Comment (strouhaljan) [inline src/VisualizationObject.ts]: Wouldn't it be better if the measure value filter is already within `VisualizationObjectFilter` interface and therefore there is no some "extended" filter `VisualizationObjectExtendedFilter`?

File: src/AFM.ts (hunk 1 above)
Comment (qduval) [reply src/AFM.ts]: Filters without condition will be ignored in gooddata-js

File: src/VisualizationObject.ts (hunk 2 above)
Comment (qduval) [reply src/VisualizationObject.ts]: This is what we meant to do at the beginning, but we noticed that some types and methods require the type IMeasureValueFilter to be excluded, for example IMeasureDefinition.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-pull-request-compliance-gate/18/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-typings-unit-tests-zuul-docker-el7/117/ : SUCCESS in 1m 05s

**Pull request merged**